app/src/main/java/com/example/abha_create_verify_android/verify/MobileLinkedABHAActivity.kt

```diff
@@ -0,0 +1,216 @@
+import com.example.abha_create_verify_android.data.api.ApiHelper
+import com.example.abha_create_verify_android.data.api.RetrofitBuilder
+import com.example.abha_create_verify_android.data.model.MobileNumberLinkedABHA
+import com.example.abha_create_verify_android.data.model.SearchAbhaReq
+import com.example.abha_create_verify_android.databinding.ActivityLinkedAbhaBinding
+import com.example.abha_create_verify_android.utils.DialogUtils
+import com.example.abha_create_verify_android.utils.Status
+import com.example.abha_create_verify_android.utils.Variables
+
+
+class MobileLinkedABHAActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityLinkedAbhaBinding
+    private lateinit var viewModel: MainViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLinkedAbhaBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setupViewModel()
+
+        setSupportActionBar(binding.appBarLayout.includeToolbar.toolbarAbha)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.title = resources.getString(R.string.verify_abha)
+
+        val dataList: ArrayList<MobileNumberLinkedABHA>? = intent.getSerializableExtra("linkedABHAs") as? ArrayList<MobileNumberLinkedABHA>
+
+        val listContainer = findViewById<LinearLayout>(R.id.listContainer)
+
+        var isABHAAddressLinked = false
+        var selectedIndex = -1
+        var selectedTextView: AppCompatTextView? = null
+
+        if (dataList != null) {
+            for ((index, dataItem) in dataList.withIndex()) {
+                val textView = createTextView(dataItem)
+
+                textView.setOnClickListener {
+                    binding.abhaAddressNotLinked.visibility = View.GONE
+                    binding.txtLinked.visibility = View.INVISIBLE
+
+                    selectedTextView?.setBackgroundColor(Color.WHITE)
+                    textView.setBackgroundResource(R.drawable.border)
+
+                    selectedTextView = textView
+                    selectedIndex = index
+
+                    if (dataList[selectedIndex].abhaAddress.isEmpty()) {
+                        binding.abhaAddressNotLinked.visibility = View.VISIBLE
+                        isABHAAddressLinked = false
+                    }
+                    else
+                    {
+                        isABHAAddressLinked = true
+                    }
+                }
+
+                listContainer.addView(textView)
+
+                val dividerView = createDividerView()
+                listContainer.addView(dividerView)
+
+            }
+        }
+
+
+        binding.proceedButton.setOnClickListener {
+            if(selectedIndex != -1 && dataList?.get(selectedIndex)  != null) {
+                if (Variables.EXISTING_ABHA_NUMBERS?.contains(dataList[selectedIndex].abhaNumber) == true) {
+                    binding.txtLinked.visibility = View.VISIBLE
```

Comment:
Here also can we set binding.txtlinked.visibility = View.VISIBLE by default

Reply:
Actually fist we have to check whether its an existing abha numbers or not. If already existing abha number, then its visible.